From a7ccb77769227e396a27261096e94efe8a196caa Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Mon, 12 Jun 2023 15:55:17 +0200
Subject: [PATCH] Disable some cross-compile checks

---
 configure.ac     | 22 ----------------------
 jri/configure.ac | 19 -------------------
 2 files changed, 41 deletions(-)

diff --git a/configure.ac b/configure.ac
index 79a175f..5d55d1e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -268,28 +268,6 @@ Make sure you have Java Development Kit installed and correctly registered in R.
 If in doubt, re-run "R CMD javareconf" as root.
 ])])
 
-AC_MSG_CHECKING([whether JNI programs run])
-AC_RUN_IFELSE([AC_LANG_SOURCE([
-#include <jni.h>
-int main(void) {
-  jsize n;
-  JNI_GetCreatedJavaVMs(NULL, 0, &n);
-  return 0;
-}
-	])],
-	[AC_MSG_RESULT([yes])],
-	[AC_MSG_ERROR([Unable to run a simple JNI program. Make sure you have configured R with Java support (see R documentation) and check config.log for failure reason.])],
-	[AC_MSG_RESULT([don't know (cross-compiling)])])
-
-AC_MSG_CHECKING([JNI data types])
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <jni.h>
-int main(void) {
-  return (sizeof(int)==sizeof(jint) && sizeof(long)==sizeof(long) && sizeof(jbyte)==sizeof(char) && sizeof(jshort)==sizeof(short) && sizeof(jfloat)==sizeof(float) && sizeof(jdouble)==sizeof(double))?0:1;
-}
-        ]])],[AC_MSG_RESULT([ok])],[AC_MSG_ERROR([One or more JNI types differ from the corresponding native type. You may need to use non-standard compiler flags or a different compiler in order to fix this.])],[])
-
-
 if test "${want_jri}" = auto; then
    AC_MSG_CHECKING([whether JRI should be compiled (autodetect)])
    AC_MSG_RESULT([${has_R_shlib}])
diff --git a/jri/configure.ac b/jri/configure.ac
index 3e56081..3377d68 100644
--- a/jri/configure.ac
+++ b/jri/configure.ac
@@ -296,25 +296,6 @@ int main(void) {
 LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${JAVA_LD_PATH0}
 export LD_LIBRARY_PATH
 
-AC_MSG_CHECKING([whether JNI programs can be run])
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <jni.h>
-int main(void) {
-    jobject o;
-    return 0;
-}
-        ]])],[AC_MSG_RESULT(yes)],
-        [AC_MSG_RESULT(no)
-	 AC_MSG_ERROR([Cannot run a simple JNI program - probably your jvm library is in non-standard location or JVM is unsupported. See config.log for details.])])
-
-AC_MSG_CHECKING([JNI data types])
-AC_RUN_IFELSE([AC_LANG_SOURCE([[
-#include <jni.h>
-int main(void) {
-  return (sizeof(int)==sizeof(jint) && sizeof(long)==sizeof(long) && sizeof(jbyte)==sizeof(char) && sizeof(jshort)==sizeof(short) && sizeof(jfloat)==sizeof(float) && sizeof(jdouble)==sizeof(double))?0:1;
-}
-        ]])],[AC_MSG_RESULT([ok])],[AC_MSG_ERROR([One or more JNI types differ from the corresponding native type. You may need to use non-standard compiler flags or a different compiler in order to fix this.])],[])
-
 JNIPREFIX=lib
 CPICF=`"${RBIN}" CMD config CPICFLAGS`
 JNISO=.so
-- 
2.39.2 (Apple Git-143)

