diff --git a/jri/configure.win b/jri/configure.win
index 757ca49..e9ced1f 100644
--- a/jri/configure.win
+++ b/jri/configure.win
@@ -42,7 +42,7 @@ else ## else we have to create headers during the compilation
     echo 'JDK has no javah.exe - using javac -h . instead'
     ## if this is at least 1.8 we can set source/target
     ## it is mandatory for 14 (and may be earlier) due to yield
-    tgt=`"${JAVA_HOME}/bin/java" -cp tools getsp -minver 8`
+    tgt=`"${PREFIX}/LIBRARY/bin/java.exe" -cp tools getsp -minver 8`
     echo "Is the Java version at least 1.8 ... $tgt"
     if [ x$tgt = xyes ]; then
        echo 'JFLAGS=-source 1.8 -target 1.8 -h .' >> src/Makefile.wconf
